src/refract/VisitorUtils.h

```diff
@@ -125,32 +125,68 @@ namespace refract
         }
     };
 
-    struct GetEnumValue {
+    template<>
+    struct GetValue<refract::EnumElement, typename refract::EnumElement::ValueType> {
+        using EnumElement = refract::EnumElement;
         const EnumElement& element;
 
-        GetEnumValue(const EnumElement& e) : element(e) {}
+        GetValue(const EnumElement& e) : element(e) {}
 
-        operator const EnumElement::ValueType*() {
+        operator const IElement*() {
+            return GetEnumValue(element);
+        }
+
+        IElement* GetEnumValue(const EnumElement& element) {
             if (const EnumElement* s = GetSample(element)) {
-                return &s->value;
+                return GetEnumValue(*s);
             }
 
             if (const EnumElement* d = GetDefault(element)) {
-                return &d->value;
+                return GetEnumValue(*d);
             }
 
-            if (!element.empty()) {
-                return &element.value;
+            if (element.value) {
+                return element.value;
             }
 
-            if (element.empty() && IsTypeAttribute(element, "nullable")) {
-                return NULL;
+            if (const ArrayElement* e = GetEnumerations(element)) {
+                if (e && !e->empty()) {
+                    for (const auto& item : e->value) {
+                        if (!item) {
```

Comment:
Maybe stupid question, is there cases where nil would be in the array and enumerating over it gives nil as a value?

Reply:
Usually not, but when it happen, we will end with SIGFAULT crash.

Reply:
In general. It array of pointers. It is really good idea to check if pointer is not NULL every time

Reply:
Especially in this case it is not error, it here not will be element. 
Enum element does not require to be "Enumerations" attribute filled or its existence